### PR TITLE
Migrate osu.Game to the VS2017 project structure

### PR DESCRIPTION
**Pull request by smoogipoo**
Requesting review from @huoyaoyuan, I need a second set of eyes to sanitise this, because I've removed a lot of references but I've tested on a clean clone and they don't seem to cause any problems. It looks like references are propagated to parents?

### CODE REVIEW

File: osu.sln [hunk 1]
```diff
@@ -34,8 +34,8 @@ Global
 		{2A66DD92-ADB1-4994-89E2-C94E04ACDA0D}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{2A66DD92-ADB1-4994-89E2-C94E04ACDA0D}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{2A66DD92-ADB1-4994-89E2-C94E04ACDA0D}.Release|Any CPU.Build.0 = Release|Any CPU
-		{2A66DD92-ADB1-4994-89E2-C94E04ACDA0D}.VisualTests|Any CPU.ActiveCfg = VisualTests|Any CPU
-		{2A66DD92-ADB1-4994-89E2-C94E04ACDA0D}.VisualTests|Any CPU.Build.0 = VisualTests|Any CPU
+		{2A66DD92-ADB1-4994-89E2-C94E04ACDA0D}.VisualTests|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (huoyaoyuan) [inline osu.sln]: If VisualTest is no longer a build config, the solution platform should be removed too

File: osu.Game/osu.Game.csproj [hunk 2]
```diff
@@ -1,827 +1,41 @@
...
   <ItemGroup>
-    <ProjectReference Include="..\osu-framework\osu.Framework\osu.Framework.csproj">
-      <Project>{c76bf5b3-985e-4d39-95fe-97c9c879b83a}</Project>
-      <Name>osu.Framework</Name>
-    </ProjectReference>
-    <ProjectReference Include="..\osu-resources\osu.Game.Resources\osu.Game.Resources.csproj">
-      <Project>{d9a367c9-4c1a-489f-9b05-a0cea2b53b58}</Project>
-      <Name>osu.Game.Resources</Name>
-    </ProjectReference>
+    <Content Include="*.so">
```
Comment (huoyaoyuan) [inline osu.Game/osu.Game.csproj]: `**\*` used in other projects. Could be better to keep consistent

File: osu.Game/osu.Game.csproj [hunk 3]
```diff
@@ -1,827 +1,41 @@
...
-    <Compile Include="Beatmaps\DifficultyCalculator.cs" />
-    <Compile Include="Beatmaps\Drawables\BeatmapBackgroundSprite.cs" />
-    <Compile Include="Beatmaps\Drawables\BeatmapGroup.cs" />
-    <Compile Include="Beatmaps\Drawables\BeatmapPanel.cs" />
-    <Compile Include="Beatmaps\Drawables\BeatmapSetCover.cs" />
-    <Compile Include="Beatmaps\Drawables\BeatmapSetHeader.cs" />
-    <Compile Include="Database\DatabaseContextFactory.cs" />
-    <Compile Include="Database\IHasPrimaryKey.cs" />
-    <Compile Include="Migrations\20171019041408_InitialCreate.cs" />
-    <Compile Include="Migrations\20171019041408_InitialCreate.Designer.cs">
```
Comment (huoyaoyuan) [inline osu.Game/osu.Game.csproj]: Could EF tools recognize these correctly?
And I'd want EF tools to generate them inside `Database` folder.

**Comment by peppy** [comment]
Please PR this again against the `netstandard` upstream branch. We need to make sure all csproj files (framework and not) are updated together to avoid complications.

**Comment by peppy** [comment]
never mind. i can do this server-side. amazing

**Comment by smoogipoo** [comment]
Closing temporarily.

**Pull request closed without merging**